src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala

```diff
@@ -346,25 +346,20 @@ class PostgresDirectoryDAO(protected val dbRef: DbReference,
   override def loadUser(userId: WorkbenchUserId): IO[Option[WorkbenchUser]] = {
     runInTransaction { implicit session =>
       val userTable = UserTable.syntax
-      val userColumn = UserTable.column
 
-      import SamTypeBinders._
-      val loadUserQuery = samsql"select ${userTable.id}, ${userTable.email}, ${userTable.googleSubjectId} from ${UserTable.table} where ${userTable.id} = ${userId}"
-      loadUserQuery.map(rs => WorkbenchUser(rs.get[WorkbenchUserId](userColumn.id), rs.stringOpt(userColumn.googleSubjectId).map(GoogleSubjectId), rs.get[WorkbenchEmail](userColumn.email)))
-        .single().apply()
+      val loadUserQuery = samsql"select ${userTable.resultAll} from ${UserTable as userTable} where ${userTable.id} = ${userId}"
+      loadUserQuery.map(UserTable(userTable))
+        .single().apply().map(unmarshalUserRecord)
     }
   }
 
   override def loadUsers(userIds: Set[WorkbenchUserId]): IO[Stream[WorkbenchUser]] = {
     runInTransaction { implicit session =>
       val userTable = UserTable.syntax
-      val userColumn = UserTable.column
-
-      import SamTypeBinders._
 
-      val loadUsersQuery = samsql"select ${userTable.id}, ${userTable.email}, ${userTable.googleSubjectId} from ${UserTable.table} where ${userTable.id} in (${userIds})"
-      loadUsersQuery.map(rs => WorkbenchUser(rs.get[WorkbenchUserId](userColumn.id), rs.stringOpt(userColumn.googleSubjectId).map(GoogleSubjectId), rs.get[WorkbenchEmail](userColumn.email)))
-        .list().apply().toStream
+      val loadUsersQuery = samsql"select ${userTable.resultAll} from ${UserTable as userTable} where ${userTable.id} in (${userIds})"
+      loadUsersQuery.map(UserTable(userTable))
+        .list().apply().map(unmarshalUserRecord).toStream
```

Comment:
the `.toStream` here is worrisome... the whole point of this returning a stream is that it does not load everything into memory.

I don't think it needs to be fixed in this PR but maybe make a JIRA to figure out how to stream results from postgres. [This](http://scalikejdbc.org/documentation/operations.html#foreach-operation) looks interesting plus the section below on Setting JDBC fetchSize.

Reply:
Agreed, https://broadworkbench.atlassian.net/browse/CA-428